src/main/java/com/nha/abdm/wrapper/hrp/serviceImpl/LogsTableService.java

```diff
@@ -100,6 +90,57 @@ public void setContent(Object content, HttpEntity<ObjectNode> requestEntity,Obje
                                         .set("rawResponse",map);
                 mongoTemplate.updateFirst(query, update, RequestLogs.class);
             }
+        }if(contentType == LinkRecordsResponse.class && Objects.nonNull(content)){
```

Comment:
If the intent here is to capture all the requests and corresponding response, then having some generic `GatewayRequest` and `GatewayResponse` should be good. Doing it case by case basis makes the code more verbose. We can discuss on this.